rdd/src/main/scala/ast/codec/RDDMamlUtilityCodecs.scala

```diff
@@ -0,0 +1,30 @@
+package maml.rdd.ast.codec
+
+import maml.rdd.eval._
+import maml.rdd.ast._
+import maml.ast._
+import maml.ast.utility._
+
+import io.circe._
+import io.circe.syntax._
+import io.circe.parser._
+import io.circe.optics.JsonPath._
+import cats.syntax.functor._
+
+import geotrellis.spark._
+
+
+trait RDDMamlUtilityCodecs {
+  implicit val rddMamlKindDecoder: Decoder[MamlKind] = Decoder[String].emap({
+    case "spatialRDD" => Right(MamlKind.Tile)
```

Comment:
It doesn't seem like these codecs can actually decode a full Expression tree, just the RDD literal. I would suggest making decoding of this a follow up PR where we need to create an explicit pattern how to extend the decoders. @moradology confirm/deny?